index.js

```diff
@@ -124,14 +124,15 @@ export default function stringifyObject(input, options, pad) {
 			return expandWhiteSpace(returnValue);
 		}
 
+		input = input.replace(/\\/g, '\\\\');
 		input = String(input).replace(/[\r\n]/g, x => x === '\n' ? '\\n' : '\\r');
 
 		if (options.singleQuotes === false) {
 			input = input.replace(/"/g, '\\"');
 			return `"${input}"`;
 		}
 
-		input = input.replace(/\\?'/g, '\\\'');
+		input = input.replace(/'/g, '\\\'');
```

Comment:
Yes. This just removes backslashes, without properly escaping them.

At least if the goal is to produce strings that can be evaluated again, i.e. `eval(stringifyObject(input)) === input`. e2b877d just steps around the issue of escape characters.

But as I said, the goal of this library may not be that... It's just that the current code produces invalid javascript for the input I presented in the description.

Reply:
Consider also the failing test (the one which said commit addresses): The input string is `'a \' b \' c \\\' d'`. At the moment, because of that, the library treats `\'` and `\\\'` the same. Is that desirable? In my opinion backslashes should be properly escaped and present in the output as appropriate, this is what this PR tries to achieve.